js/h5p.js

```diff
@@ -224,10 +224,16 @@ H5P.init = function (target) {
         instance.getCurrentState instanceof Function ||
         typeof instance.getCurrentState === 'function')) {
 
-      var saveTimer, save = function () {
+      var saveTimer, lastSavedState, save = function () {
         var state = instance.getCurrentState();
-        if (state !== undefined) {
+
+        //compare last state with current state
+        var stateChanged = H5P.eq(state, lastSavedState, null, null);
+        
+        //setUserData and save only when state is defined and the state has changed
```

Comment:
The comments, in general, are a little bit lower quality than I'd like to see, but I'll try to pick apart this one in particular.

- Try to adopt the project's general coding style.  In this case, that means a space after the "//" and a comment beginning with a capital letter.  In general, your contributions are slightly more verbose than the project as a whole.
- Describe the business need being solved rather than the code's behavior
- Be as concise as possible
- Don't comment or annotate trivial or easily understood code

From what moment do we measure the state as having been changed?  Why are we using two verbs ("setUserData" and "save") to describe a single side-effect of this block, when we're already in the "save" function?

You're close, but a cleanup pass on all comments, probably removing a few, needs to happen.